src/utils/assets.js

```diff
@@ -19,24 +19,37 @@
 */
 import { utils } from 'ethers';
 import { BigNumber } from 'bignumber.js';
-import type { Assets, Balances, Rates } from 'models/Asset';
+import type {
+  Asset,
+  Assets,
+  Balance,
+  Balances,
+  Rates,
+} from 'models/Asset';
 import get from 'lodash.get';
 import { ETH } from 'constants/assetsConstants';
 import { formatAmount, isCaseInsensitiveMatch } from 'utils/common';
 
-export function transformAssetsToObject(assetsArray: Object[] = []): Object {
+export const transformAssetsToObject = (assetsArray: Asset[] = []): Assets => {
   return assetsArray.reduce((memo, asset) => {
     memo[asset.symbol] = asset;
     return memo;
   }, {});
-}
+};
+
+export const getBalance = (balances: Balances, asset: string): number => {
+  const assetBalance = balances[asset];
+
+  if (!assetBalance) {
+    return 0;
+  }
+
+  const number = new BigNumber(assetBalance.balance);
 
-export function getBalance(balances: Balances = {}, asset: string = ''): number {
-  const number = balances[asset] ? new BigNumber(balances[asset].balance) : 0;
   return +formatAmount(number.toString());
-}
+};
 
-export function getRate(rates: Rates = {}, token: string, fiatCurrency: string): number {
```

Comment:
Now it became less safe :)
How about using getters for `tokenRates` and `ethRate` ?

Reply:
Do you think we're ever passing `rates = undefined` here? Flow should prevent that, since the argument is not optional :thinking:

Reply:
We're fetching the rates from external api and never know what it returns back :)
Probably the ` = {}` was put here because we already had this sort of issues

Reply:
Okay, I understand it's probably not the best time to change this. Let's see if we can improve it later.

Reply:
Updated.